### PR TITLE
Add a dangling image prefix when tag name is not specified.

### PR DESCRIPTION
**Pull request by manugupt1**
Fixes: https://github.com/containerd/nerdctl/issues/1398

### CODE REVIEW

**Comment by manugupt1** [comment]
@AkihiroSuda can I get a review on this please.
Thanks!

File: cmd/nerdctl/build_test.go [hunk 1]
```diff
@@ -402,3 +403,31 @@ CMD ["echo", "dockerfile"]
...
+
+	base.Cmd("--debug", "build", buildCtx).AssertOK()
+	imgs := base.Cmd("images", "--format", "json").OutLines()
+	imgs = imgs[:len(imgs)-1]
+	var noneImage *imagePrintable
+	for _, img := range imgs {
+		err := json.Unmarshal([]byte(img), &noneImage)
+		assert.NilError(t, err, img)
+		if noneImage.Tag == "<none>" {
+			break
```
Comment (fahedouch) [inline cmd/nerdctl/build_test.go:426]: `continue` to check the hole list of `imgs`

File: cmd/nerdctl/build_test.go [hunk 2]
```diff
@@ -402,3 +403,31 @@ CMD ["echo", "dockerfile"]
...
+		}
```
Comment (fahedouch) [inline cmd/nerdctl/build_test.go:427]: To check  that at least we have an image with tag  `<none>`  you have to provide a bool before the break
```
if noneImage.Tag == "<none>" {
                        nonTagExists = true
			break
		}
```

Comment (fahedouch) [reply cmd/nerdctl/build_test.go:427]: Yes I got the point of `noneImage ` but it is more reader friendly with bool

**Comment by ktock** [comment]
@manugupt1 Doesn't https://github.com/moby/buildkit/issues/3212 affect this implementation?

**Comment by manugupt1** [comment]
> @manugupt1 Doesn't [moby/buildkit#3212](https://github.com/moby/buildkit/issues/3212#issuecomment-1288375930) affect this implementation?

It affects only scratch images or images with no layer. https://github.com/moby/buildkit/issues/3212#issuecomment-1288375930 ; unpacking should be skipped for when there are no layers I believe is the right solution in buildkit.

**Comment by manugupt1** [comment]
@ktock PTAL, this and the PR (https://github.com/moby/buildkit/pull/3251)  in buildkit should fix all the concerns so far. LMK if you have questions.

File: cmd/nerdctl/build_test.go [hunk 3]
```diff
@@ -402,3 +403,33 @@ CMD ["echo", "dockerfile"]
 	base.Cmd("build", "-t", imageName, buildCtx).AssertOK()
 	base.Cmd("run", "--rm", imageName).AssertOutExactly("dockerfile\n")
 }
+
+func TestBuildNoTag(t *testing.T) {
+	testutil.RequiresBuild(t)
+	testutil.DockerIncompatible(t)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/build_test.go:409]: Why incompatible?

**Review by ktock — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**